src/main/java/de/picturesafe/search/elasticsearch/impl/SingleIndexElasticsearchServiceImpl.java

```diff
@@ -30,13 +31,13 @@
 
     @Autowired
     public SingleIndexElasticsearchServiceImpl(ElasticsearchService elasticsearchService,
-                                               IndexPresetConfigurationProvider indexPresetConfigurationProvider) {
+                                               IndexPresetConfigurationProvider indexPresetConfigurationProvider,
+                                               @Value("${elasticsearch.index.alias:default}") String defaultAlias) {
```

Comment:
I would not make it mandatory since there is a default ("default"). And when using the lib outside of the spring environment this constructor seems a bit strange.

Reply:
This is not possible because the setter injection takes place after the constructor call. Access to the IndexPresetConfigurationProvider is now dynamic.